third_party/docfx-doclet-143274/src/main/java/com/microsoft/util/Utils.java

```diff
@@ -250,4 +252,19 @@ public List<Element> getImplementedMethods(
   public List<TypeElement> getImplementedInterfaces(TypeElement element) {
     return element.getInterfaces().stream().map(e -> asTypeElement(e)).collect(Collectors.toList());
   }
+
+  // Use to get the recommended package URL for Package Overview
+  public static String[] extractPackageBaseURIBeforeVersion(String input) {
+    // This regex captures the package path before the version (if it exists) and the version
+    Pattern pattern = Pattern.compile("(.*?)(v\\d+.*?)(?:\\.|$)");
+    Matcher matcher = pattern.matcher(input);
+    boolean isVersioned = matcher.find();
+    if (isVersioned) {
+      String[] packageInfo = {matcher.group(1), matcher.group(2)};
+      return packageInfo;
+    } else {
+      String[] packageInfo = {"N/A", "N/A"};

```

Comment:
If this method is intended to be used as a generic utility, it shouldn't assume that `N/A` is the desired representation of "nothing found". If it's intended for a specific use, place this method with the class that needs the specific behavior.

Reply:
Makes sense - it is only used for the Package Overview so I moved it within that class.